docs/examples/vector_stores/existing_data/pinecone_existing_data.ipynb

```diff
@@ -0,0 +1,385 @@
+    "  },\n",
+    "  {\n",
+    "    \"title\": \"Pride and Prejudice\",\n",
+    "    \"author\": \"Jane Austen\",\n",
+    "    \"content\": \"Pride and Prejudice is a novel by Jane Austen published in 1813...\",\n",
+    "    \"year\": 1813\n",
+    "  },\n",
+    "]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2d96f974-39cb-4388-a315-e73b7ccf01db",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "### Add data\n",
+    "We add the sample books to our Weaviate \"Book\" class (with embedding of content field"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "id": "1c6aa5bf-466f-4a08-913f-18b65f2e75dd",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "import uuid\n",
+    "from llama_index.embeddings.openai import OpenAIEmbedding\n",
+    "embed_model = OpenAIEmbedding()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "id": "51baeaa1-9bba-4421-b978-8fa0bfafe4f3",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "{'upserted_count': 4}"
+      ]
+     },
+     "execution_count": 22,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "entries = []\n",
+    "for book in books:\n",
+    "    vector = embed_model.get_text_embedding(book['content'])\n",
+    "    entries.append(\n",
+    "        {\n",
+    "            'id': str(uuid.uuid4()),\n",
+    "            'values': vector,\n",
+    "            'metadata': book\n",
+    "        }\n",
+    "    )\n",
+    "pinecone_index.upsert(entries)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f7d26d83-dc3d-4b07-9faa-da3246ee46eb",
+   "metadata": {},
+   "source": [
+    "## Query Against \"Existing\" Pinecone Vector Store "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "bae39494-5c9f-43e7-8b26-f1afbd62c60a",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "from llama_index.vector_stores import PineconeVectorStore\n",
+    "from llama_index import VectorStoreIndex\n",
+    "from llama_index.response.pprint_utils import pprint_source_node"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8647c689-5964-446f-9e5e-1da620483cef",
+   "metadata": {},
+   "source": [
+    "You must properly select a class property as the \"text\" field."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "id": "59093426-0fda-46d0-84b5-8b84d6a81fe8",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "vector_store = PineconeVectorStore(pinecone_index=pinecone_index, text_key='content')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "d8f06305-726b-45b8-bca7-e30e6d84524a",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "retriever = VectorStoreIndex.from_vector_store(vector_store).as_retriever(similarity_top_k=1)"
```

Comment:
Does this only work with retrievers? Or will the query engine work too?

Reply:
Both should work.

Only showing retriever here, since I was going for the philosophy of keeping the notebooks simple, showcasing a specific module, rather than an overall tutorial for a full use-case.

Not sure if that really comes across though.